ouroboros-network/src/Ouroboros/Network/Mux/Interface.hs

```diff
@@ -124,12 +94,12 @@ data NetworkInterface ptcl addr m r = NetworkInterface {
       -- listen for incoming connections.  Some bearers do not have a notion of
       -- address.
       --
-      nodeAddress      :: addr,
+      nodeAddress     :: addr,
```

Comment:
It'd be nice if we can pull this out. It's only needed for the `MuxApplication`s that have a server side to them.

Perhaps we don't need a `NetworkInterface` at all. Instead we can have run functions for `MuxApplication`. Though might need a GADT or phantom type to make sure that a `runMuxApplication` matches up with the runtime args needed to run (i.e. an addr for the server cases).